shadows/framework/src/main/java/org/robolectric/android/controller/ActivityController.java

```diff
@@ -53,6 +54,13 @@
   private _Activity_ _component_;
 
   public static <T extends Activity> ActivityController<T> of(T activity, Intent intent) {
+    if (intent != null) {
```

Comment:
Can you move this into a private method in the base class (ComponentController)? e.g. call it something like `validateIntent` or `parcelAndUnparcelIntent`. It should be called in ComponentController.of as well as `ActivityController.onNewIntent`

Reply:
This way the logic gets reused for ServiceController as well.

Reply:
Sure, I forgot about ServiceController.